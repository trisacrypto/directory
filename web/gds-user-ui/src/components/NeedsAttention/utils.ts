export const enum AttentionAction {
  'VERIFY_EMAILS' = 'VERIFY_EMAILS',
  'VERIFY_PHONES' = 'VERIFY_PHONES',
  'VERIFY_ADDRESSES' = 'VERIFY_ADDRESSES',
  'VERIFY_IDENTITIES' = 'VERIFY_IDENTITIES',
  'VERIFY_DOCUMENTS' = 'VERIFY_DOCUMENTS',
  'VERIFY_ALL' = 'VERIFY_ALL',
  'VERIFY_NONE' = 'VERIFY_NONE'
}
