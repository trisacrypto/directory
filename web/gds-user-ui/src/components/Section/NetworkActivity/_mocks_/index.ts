export const mockNetworkActivityData: any = {
    networkActivity: {
            mainnet: [
                { date: "2023-08-14", events: 5 },
                { date: "2023-08-15", events: 30 },
                { date: "2023-08-16", events: 10 },
                { date: "2023-08-17", events: 4 },
                { date: "2023-08-18", events: 40 },
                { date: "2023-08-21", events: 6 },
                { date: "2023-08-22", events: 2 },
                { date: "2023-08-23", events: 15 },
                { date: "2023-08-24", events: 1 },
                { date: "2023-08-25", events: 8 },
                { date: "2023-08-28", events: 13 },
                { date: "2023-08-29", events: 30 },
                { date: "2023-08-30", events: 12 },
                { date: "2023-08-31", events: 8 },
                { date: "2023-09-01", events: 2 },
            ],
            testnet: [
                { date: "2023-08-14", events: 2 },
                { date: "2023-08-15", events: 10 },
                { date: "2023-08-16", events: 30 },
                { date: "2023-08-17", events: 8 },
                { date: "2023-08-18", events: 25 },
                { date: "2023-08-21", events: 12 },
                { date: "2023-08-22", events: 24 },
                { date: "2023-08-23", events: 50 },
                { date: "2023-08-24", events: 18 },
                { date: "2023-08-25", events: 38 },
                { date: "2023-08-28", events: 3 },
                { date: "2023-08-29", events: 40 },
                { date: "2023-08-30", events: 1 },
                { date: "2023-08-31", events: 3 },
                { date: "2023-09-01", events: 9 },
            ],
    },
 };

