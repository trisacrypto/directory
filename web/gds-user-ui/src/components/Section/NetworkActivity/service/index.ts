import axiosInstance from "utils/axios";

export const networkActivity = async () => {
    const res = await axiosInstance.get(`/network/activity`);
    return res.data;
};
