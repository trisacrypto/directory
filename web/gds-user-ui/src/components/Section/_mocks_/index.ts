export const mockNetworkActivityData: any = {
    networkActivity: {
            mainnet: [
                { x: "2023-08-14", y: 5 },
                { x: "2023-08-15", y: 30 },
                { x: "2023-08-16", y: 10 },
                { x: "2023-08-17", y: 4 },
                { x: "2023-08-18", y: 40 },
                { x: "2023-08-21", y: 6 },
            ],
            testnet: [
                { x: "2023-08-14", y: 2 },
                { x: "2023-08-15", y: 10 },
                { x: "2023-08-16", y: 30 },
                { x: "2023-08-17", y: 8 },
                { x: "2023-08-18", y: 25 },
                { x: "2023-08-21", y: 12 },
            ],
    },
 };

