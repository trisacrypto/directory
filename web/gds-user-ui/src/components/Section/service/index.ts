import axiosInstance from "utils/axios";

export const networkActivity = async () => {
    const response = await axiosInstance.get(`/network/activity`);
    return response.data;
};
