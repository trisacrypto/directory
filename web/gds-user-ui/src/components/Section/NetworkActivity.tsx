import { VStack } from '@chakra-ui/react';
import { VictoryAxis, VictoryChart, VictoryLegend, VictoryLine, VictoryTheme } from 'victory';
import { mockNetworkActivityData } from './_mocks_';

const legendData = [
    { name: "TestNet", symbol: { fill: "black" } },
    { name: "MainNet", symbol: { fill: "#F1511B" } },
];

const mainnetData = mockNetworkActivityData?.networkActivity.mainnet;
const testnetData = mockNetworkActivityData?.networkActivity.testnet;

const NetworkActivity = () => {
    return (
        <>
        <VStack maxW={'5xl'}>
        <VictoryChart domainPadding={20} style={{ background: { fill: "#F7F9FB" } }} theme={VictoryTheme.material} width={600}>
            <VictoryLegend data={legendData} title="Network Activity" orientation="horizontal" titleOrientation="left" />
            <VictoryAxis tickValues={mainnetData.map((x: any) => x.x)} />
            <VictoryAxis dependentAxis />
            <VictoryLine data={testnetData} style={{ data: { stroke: "black" } }} />
            <VictoryLine data={mainnetData} style={{ data: { stroke: "#F1511B" } }}/>
        </VictoryChart>
        </VStack>
        </>
    );
};

export default NetworkActivity;
