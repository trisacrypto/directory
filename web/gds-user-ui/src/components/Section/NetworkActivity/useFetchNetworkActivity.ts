import { useState } from "react";
import { networkActivity } from "./service";

const useFetchNetworkActivity = () => {
    const [isLoading, setIsLoading] = useState(false);
    const [data, setData] = useState<any>(null);
    const [error, setError] = useState<any>(null);
    
    const fetchNetworkActivity = async () => {
        setIsLoading(true);
        try {
            const res = await networkActivity();
            if (!res.mainnet && !res.testnet) setError("No network activity found.");
            setData(res);
        } catch (e: any) {
            if (!e?.res?.data?.success) {
                setError(e?.res?.data?.error);
            } else {
                setError("Something went wrong.");
            }
        } finally{
            setIsLoading(false);
        }
    };
    fetchNetworkActivity();
    return { data, isLoading, error };
};

export default useFetchNetworkActivity;
