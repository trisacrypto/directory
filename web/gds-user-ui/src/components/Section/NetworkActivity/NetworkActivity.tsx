import { VStack } from '@chakra-ui/react';
import { VictoryAxis, VictoryChart, VictoryLegend, VictoryLine, VictoryTheme } from 'victory';
import formatDisplayedDate from 'utils/formatDisplayedDate';
import useFetchNetworkActivity from './useFetchNetworkActivity';

const NetworkActivity = () => {
    const { data } = useFetchNetworkActivity();
    const mainnetData = data?.mainnet;
    const testnetData = data?.testnet;
    
    // The victory time scale requires dates to be in a Date object.
    mainnetData?.forEach((date: any) => {
      date.x = formatDisplayedDate(date.x);
    });
    
    testnetData?.forEach((date: any) => {
      date.x = formatDisplayedDate(date.x);
   });
   
   const legendData = [
    { name: "TestNet", symbol: { fill: "black" } },
    { name: "MainNet", symbol: { fill: "#F1511B" } }
  ];
  
  // Add padding to the axis labels to prevent overlap with the axis ticks.
  const sharedAxisStyles = {
    axisLabel: { padding: 35, fontWeight: 500, color: "black" },
  };

  return (
  <section>
    <VStack maxW={'5xl'} margin="auto" marginTop={10}>
      <VictoryChart
        domain={{ y: [0, 50] }}
        domainPadding={{ x: 1, y: 20 }}
        width={600} 
        style={{ background: { fill: "#F7F9FB" } }} 
        theme={VictoryTheme.material}
        scale={{ x: "time", y: "linear" }}
        >
          <VictoryLegend 
            data={legendData} 
            orientation="horizontal" 
            x={225}
            gutter={20}
            />
            <VictoryAxis fixLabelOverlap={true} />
            <VictoryAxis dependentAxis label="Network Activity" style={sharedAxisStyles} />
            <VictoryLine 
              data={testnetData} 
              x="date" 
              y="events"
              style={{ data: { stroke: "black" } }} 
              />
              <VictoryLine 
                data={mainnetData} 
                x="date" 
                y="events"
                style={{ data: { stroke: "#F1511B" } }}
                />
      </VictoryChart>
    </VStack>
  </section>
  );
};

export default NetworkActivity;
