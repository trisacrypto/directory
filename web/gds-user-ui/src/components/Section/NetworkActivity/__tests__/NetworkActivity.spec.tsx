import axios from "axios";
import { mockNetworkActivityData } from "../_mocks_";
import { networkActivity } from "../service";
import { render, screen } from "utils/test-utils";
import NetworkActivity from "../NetworkActivity";

function renderComponent() {
    return render(<NetworkActivity />);
}

describe('Network Activity Service', () => {
    it('returns network service data with response', async () => {
        const data = mockNetworkActivityData;
        axios.get = jest.fn().mockResolvedValue({ data });
        await expect (networkActivity()).resolves.toEqual(data);
    });
});

describe('Network Activity Component', () => {
    it('should render', () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
    });

/*     it('should display the correct label on the y-axis', () => {
        renderComponent();
        const label = screen.getByDisplayValue('Network Activity');
        expect(label).toBeInTheDocument();
    });

    it('should display the testnet data', async () => {
        renderComponent();
        const { data } = mockNetworkActivityData;
       const testnetData = data?.networkActivity?.testnet
        expect(testnetData).toBeInTheDocument();
    });

    it('should display the mainnet data', async () => {
        renderComponent();
        const { data } = mockNetworkActivityData;
        const mainnetData = data?.networkActivity?.mainnet
        expect(mainnetData).toBeInTheDocument();
    }); */
});