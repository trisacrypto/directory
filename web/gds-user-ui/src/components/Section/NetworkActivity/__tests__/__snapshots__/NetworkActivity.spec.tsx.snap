// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Network Activity Component should render 1`] = `<div />`;
